### PR TITLE
Subscriptions fixes

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
This PR sets the maximum event listeners to infinite in subscriptions plugins to avoid warning on events leak (this is expected behaviour). Proper cleanup tests added as well

### CODE REVIEW

File: packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        await wsClient2.close();
+    });
+
+    afterAll(async () => {
+        await server.close();
+        await driver.close();
+    });
+
+    // NOTE: This test **may** be flaky, if so, feel free to remove it
+    test("listeners are properly triggered and cleaned up", async () => {
```
Comment (tbwiss) [inline packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts:89]: question: Would adding a delay of 100ms before checking the listenerCount make the test less flaky?

File: packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.fieldNames.subscriptions.created]: { title: "movie2" },
+                    event: "CREATE",
+                    timestamp: expect.any(Number),
+                },
+            },
+        ]);
+    });
+
+    async function createMovie(title: string): Promise<Response> {
```
Comment (tbwiss) [inline packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts:157]: question (absolutely non-blocking): Do you think there is any value in reusing this function for all the E2E subscription tests (delete, update, create)? It seems to be present in most of the E2E tests in a very similar form.

File: packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            {
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.operations.subscribe.payload.created]: { title: "movie1" },
+                    event: "CREATE",
+                    timestamp: expect.any(Number),
+                },
+            },
+            {
+                [typeMovie.operations.subscribe.created]: {
+                    [typeMovie.fieldNames.subscriptions.created]: { title: "movie2" },
```
Comment (tbwiss) [inline packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts:149]: question: Shouldn't this also be `[typeMovie.operations.subscribe.payload.created]: `?

**Review by tbwiss — APPROVED** [review]
LGTM @angrykoala!

File: packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts:89]: The flakiness should be fixed by the 50ms delay that is already in place (the comment refers to those). In my tests, I didn't see any problem in the first check

File: packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts (hunk 2 above)
Comment (angrykoala) [reply packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts:157]: I thought about that, but in the end this is a concrete query, dependant on the test, it already reuses all the "make a query" logic, so I think is better to leave it as-is for now

File: packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts (hunk 3 above)
Comment (angrykoala) [reply packages/graphql/tests/e2e/subscriptions/single-instance-plugin.e2e.test.ts]: Good question. mmm, I think we have a bit of duplicate code here. Both seem to be exactly the same. I think the alternative you propose is better, so I will remove the "fieldNames" property and change all references

**Pull request merged**